__main__.py

```diff
@@ -5,36 +5,109 @@
+                          and createNextEvent
+    input(dict):          Dict object representing a parsed cumulus message
+
+    Returns:
+    result: JSON response to pass to the next event
+    """
     if 'schemas' in allInput:
         schemas = allInput['schemas']
     else:
         schemas = None
     transformer = MessageAdapter(schemas)
-    exitCode = 1
     event = allInput['event']
+    context = allInput.get('context')
+    if functionName == 'loadAndUpdateRemoteEvent':
+        result = transformer.load_and_update_remote_event(event, context)
+    elif functionName == 'loadNestedEvent':
+        result = transformer.load_nested_event(event, context)
+    elif functionName == 'createNextEvent':
+        handlerResponse = allInput['handler_response']
+        if 'message_config' in allInput:
+            messageConfig = allInput['message_config']
+        else:
+            messageConfig = None
+        result = transformer.create_next_event(handlerResponse, event, messageConfig)
+    return result
 
-    try:
-        context = allInput.get('context')
-        if functionName == 'loadAndUpdateRemoteEvent':
-            result = transformer.load_and_update_remote_event(event, context)
-        elif functionName == 'loadNestedEvent':
-            result = transformer.load_nested_event(event, context)
-        elif functionName == 'createNextEvent':
-            handlerResponse = allInput['handler_response']
-            if 'message_config' in allInput:
-                messageConfig = allInput['message_config']
-            else:
-                messageConfig = None
-            result = transformer.create_next_event(handlerResponse, event, messageConfig)
-        if (result is not None and len(result) > 0):
-            sys.stdout.write(json.dumps(result))
+
+def streamCommands():
+    """
+    Method that runs, and reads messages on STDIN in the format:
+
+    FunctionName
+    JSON string
+    <EOC>
+
+    Method writes responses back to STDOUT in the following format:
+
+    JSON string
+    <EOC>
+
+    A single line "<EXIT>" input will cause the program to exit
+    """
+
+    cont = True
+    buffer = ''
+    command = ''
+    jsonObj = {}
+    while cont:
+        next_line = sys.stdin.readline().rstrip('\n')
+        if next_line == '<EXIT>':
+            cont = False
+        elif next_line == '<EOC>':
+            jsonObj = json.loads(buffer)
+            result = callMessageAdapterFunction(command, jsonObj)
+            sys.stdout.write(json.dumps(result) + "\n")
+            sys.stdout.write('<EOC>\n')
             sys.stdout.flush()
+            buffer = ''
+            command = ''
+        else:
+            if not command:
+                command = next_line.strip()
+                sys.stderr.write(f'warning setting command to {command}\n')
+            else:
+                buffer += next_line
+
+
+def singleCommand(functionName):
+    """Executes a single CMA command"""
+    allInput = json.loads(input())
```

Comment:
So you can still run a single command using the old style of input where the JSON has to be all on one line?

Reply:
Indeed, I just tried this